main.go

```diff
@@ -129,44 +257,75 @@ func (c *CASProxy) ValidateTicket(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	username := string(fields[1])
+
 	// Store a session, hopefully to short-circuit the CAS redirect dance in later
 	// requests. The max age of the cookie should be less than the lifetime of
 	// the CAS ticket, which is around 10+ hours. This means that we'll be hitting
 	// the CAS server fairly often. Adjust the max age to rate limit requests to
 	// CAS.
-	err = session.PutInt(r, sessionKey, 1)
+	var s *sessions.Session
+	s, err = c.sessionStore.Get(r, sessionName)
 	if err != nil {
-		err = errors.Wrap(err, "error setting setting value")
+		err = errors.Wrap(err, "error getting session")
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	s.Values[sessionKey] = username
+	s.Save(r, w)
 
 	http.Redirect(w, r, svcURL.String(), http.StatusFound)
 }
 
+// ResetSessionExpiration should reset the session expiration time.
+func (c *CASProxy) ResetSessionExpiration(w http.ResponseWriter, r *http.Request) error {
+	session, err := c.sessionStore.Get(r, sessionName)
+	if err != nil {
+		return err
+	}
+
+	msg, ok := session.Values[sessionKey]
+	if !ok {
+		return errors.New("session value not found")
+	}
+
+	session.Values[sessionKey] = msg.(string)
```

Comment:
I'm not quite clear on how this works, can you explain this syntax tidbit?

Reply:
https://tour.golang.org/methods/15 since I saw this :)

Reply:
which is to say, msg is probably an `interface{}` so you gotta pull out the real value